### PR TITLE
Apply fixes for onBuildComplete and route module

### PR DESCRIPTION
**Pull request by ijjk**
This applies fixes for outputs in `onBuildComplete` with various slight differences from expected values. This also applies some fixes in the route-module interface to ensure we install the process handlers for PPR since currently that was done in `next-server` which we want to get rid of and there was an edge case with query parsing logic in the route-module. 

Our E2E tests were run against this here to validate https://github.com/vercel/next.js/actions/runs/21324578779 and failures are matching existing failures on canary. 

x-ref: https://github.com/nextjs/adapter-vercel/pull/11
x-ref: https://github.com/vercel/vercel/pull/14574

### CODE REVIEW

File: packages/next/src/server/route-modules/route-module.ts [hunk 1]
```diff
@@ -880,6 +896,11 @@ export abstract class RouteModule<
     for (const key of routeParamKeys) {
       if (!(key in originalQuery)) {
         delete query[key]
+        // handle the case where there's collision and we
+        // normalized nxtPid=123 -> id=123 but user also
+        // sends id=456 as separate key
+      } else if (originalQuery[key] !== query[key]) {
```
[the remark this answers is not in the document]
Comment (ijjk) [reply packages/next/src/server/route-modules/route-module.ts]: This is not accurate

**Review by feedthejim — APPROVED** [review]
